pkg/util/utils.go

```diff
@@ -244,3 +270,106 @@ func UpdateK8sResourceAnnotation(client client.Client, ctx *context.Context, k8s
 	}
 	return nil
 }
+
+// some resources has complex index, so set it type to string
+func GenerateID(res_id, prefix, suffix string, index string) string {
```

Comment:
func GenerateID(res_id string, prefix string, suffix string, index string) string {

Reply:
<img width="824" alt="image" src="https://github.com/vmware-tanzu/nsx-operator/assets/4385934/8e6d6473-2e1a-4384-a2e7-2ed73f39e9da">
it's better to comply with go doc format.

Reply:
done